embed/script.js

```diff
@@ -0,0 +1,252 @@
+function h() {
+  var type = arguments[0] || 'div';
+  var attributes = arguments[1] || {};
+  var children = Array.prototype.slice.call(arguments, 2);
+  var el = document.createElement(type);
+
+  if (children[0] instanceof Array) {
+    children = children[0];
+  }
+
+  Object.keys(attributes)
+    .map(function(key) {
+      return [key, attributes[key]];
+    })
+    .forEach(function(data) {
+      var key = data[0],
+        value = data[1];
+      if (typeof value === 'function') {
+        el.addEventListener(
+          key,
+          function(e) {
+            value(e);
+          },
+          false
+        );
+      } else {
+        el.setAttribute(key, value);
+      }
+    });
+
+  if (children instanceof Array) {
+    appendChildren(el, children);
+  } else if (children instanceof HTMLElement) {
+    el.appendChild(children);
+  } else if (typeof children === 'string') {
+    el.innerText = children;
+  }
+
+  return el;
+}
+
+function appendChildren(el, children) {
+  return children
+    .filter(function(child) {
+      return child !== null;
+    })
+    .forEach(function(child) {
+      if (child instanceof HTMLElement) {
+        el.appendChild(child);
+      } else if (child instanceof Array) {
+        appendChildren(el, child);
+      } else {
+        el.innerHTML += child;
+      }
+    });
+}
+
+(function() {
+  function addControls() {
+    const $root = document.querySelector('#sttm-root');
+    const $controls = $root.querySelector('.sttm-shabad-controls');
+
+    const $transliteration = Array.prototype.slice.call(
```

Comment:
This whole block so ideally be wrapped in a:

```
if ($root) {
    // ...
}
```

Reply:
Ideally yes, however `$root` is totally owned by us.